agents/agents/guard/fraud.go

```diff
@@ -21,6 +24,7 @@ func (g Guard) handleSnapshot(ctx context.Context, log ethTypes.Log) error {
 
 	// Verify each state in the snapshot.
 	for stateIndex, state := range fraudSnapshot.Snapshot.States() {
+		stateIndex, state := stateIndex, state
```

Comment:
This is to avoid value mirroring when we dispatch to the submitter right? Do you think it's more readable to change the loop variable names? Could go either way really, I'm just being a nit :)

Something like:

```go
	for i, s := range fraudSnapshot.Snapshot.States() {
		stateIndex, state := i, s
```

Reply:
agreed!